patches/server/1044-itemstack-expose-tooltip-lines.patch

```diff
@@ -0,0 +1,30 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Yannick Lamprecht <yannicklamprecht@live.de>
+Date: Mon, 22 Jan 2024 13:27:30 +0100
+Subject: [PATCH] itemstack expose tooltip lines
+
+
+diff --git a/src/main/java/org/bukkit/craftbukkit/util/CraftMagicNumbers.java b/src/main/java/org/bukkit/craftbukkit/util/CraftMagicNumbers.java
+index 509f8487a170f3dc84b091acf16df26e42391189..8eb18f2e847fcb61fc06044be48274c6dc7f12f6 100644
+--- a/src/main/java/org/bukkit/craftbukkit/util/CraftMagicNumbers.java
++++ b/src/main/java/org/bukkit/craftbukkit/util/CraftMagicNumbers.java
+@@ -641,6 +641,19 @@ public final class CraftMagicNumbers implements UnsafeValues {
+         return org.bukkit.craftbukkit.CraftStatistic.getNMSStatistic(statistic).getName();
+     }
+     // Paper end
++    // Paper start - expose itemstack tooltip lines
++    @Override
++    public List< net.kyori.adventure.text.Component> tooltipLines(ItemStack itemStack, final org.bukkit.entity.Player player, final org.bukkit.inventory.TooltipOptions options) {
++        Preconditions.checkArgument(options != null, "options == null");
++        net.minecraft.world.item.TooltipFlag flag = switch (options){
++            case NORMAL -> net.minecraft.world.item.TooltipFlag.NORMAL;
++            case ADVANCED -> net.minecraft.world.item.TooltipFlag.ADVANCED;
++            case NORMAL_CREATIVE -> net.minecraft.world.item.TooltipFlag.NORMAL.asCreative();
++            case ADVANCED_CREATIVE -> net.minecraft.world.item.TooltipFlag.ADVANCED.asCreative();
++        };
++        List<net.minecraft.network.chat.Component> lines = CraftItemStack.asNMSCopy(itemStack).getTooltipLines(player == null ? null: ((org.bukkit.craftbukkit.entity.CraftPlayer)player).getHandle(), flag);
```

Comment:
You should use `CraftItemStack#unwrap` here instead of copy. There is no need to copy the stack as there shouldn't be any mutation of the stack going on (I think).

Reply:
I changed it back to asNMSCopy because apparently there is some possibility of a mutation in the tooltip lines (when an exception is caught parsing the JSON for display name and lore, it removes the display name and lore).